core/jvm/src/main/scala/bobcats/HashPlatform.scala

```diff
@@ -16,19 +16,39 @@
 
 package bobcats
 
-import cats.effect.kernel.Async
+import java.security.{NoSuchAlgorithmException, Security}
+import fs2.{Pipe, Stream}
+import cats.effect.kernel.{Async, Sync}
 import scodec.bits.ByteVector
 
-import java.security.MessageDigest
-
 private[bobcats] trait HashCompanionPlatform {
-  implicit def forAsync[F[_]](implicit F: Async[F]): Hash[F] =
+
+  private[bobcats] def forSync[F[_]](implicit F: Sync[F]): Hash[F] = {
+    // TODO: What to do with this?
+    val providers = Security.getProviders().clone()
```

Comment:
So I'm pretty happy with the behaviour of `Hash1` now, but if we wrap `Hash` in another `F[_]` then you can't provide it globally. That might not be too bad; we're essentially saying that you need to initialize `Hash` before you start using it.

OTOH, this might be a bit clumsy.

Reply:
> but if we wrap `Hash` in another `F[_]` then you can't provide it globally. That might not be too bad; we're essentially saying that you need to initialize `Hash` before you start using it.

I think this is the right choice. If we want to implement pooling this will be necessary anyway.